src/Common/src/Interop/Unix/libssl/SslConnectionInfo.cs

```diff
@@ -190,5 +202,32 @@ private static HashAlgorithmType MapHashAlgorithmType(Interop.libssl.DataHashAlg
                     return HashAlgorithmType.None;
             }
         }
+        
+        private static int GetHashAlgorithmBitSize(Interop.libssl.DataHashAlgorithm mac)
+        {
+            switch (mac)
+            {
+                case Interop.libssl.DataHashAlgorithm.SSL_MD5:
+                    return MD5_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_SHA1:
+                    return SHA1_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_GOST94:
+                    return GOST_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_GOST89MAC:
+                    return GOST_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_SHA256:
+                    return SHA256_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_SHA384:
+                    return SHA384_HashKeySize;
+
+                default:
+                    return 0;
```

Comment:
We should be sure this is doing something sensible on OSX.

[Our struct](https://github.com/dotnet/corefx/blob/2c6a6dc47ce711ced3705d09641a816d4903baa5/src/Common/src/Interop/Unix/libssl/Interop.libssl.cs) looks like [OpenSSL Master](https://github.com/openssl/openssl/blob/master/ssl/ssl_locl.h#L538-L555), but might be different than [OpenSSL 0.9.8zg](https://github.com/openssl/openssl/blob/OpenSSL_0_9_8zg/ssl/ssl.h#L354-L365) (which is what OSX has).  I'm not sure that we're doing anything sensible on OSX.